controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -130,25 +139,43 @@ func (r *StorageConsumerReconciler) initReconciler(request reconcile.Request) {
 
 	r.cephClientRBDProvisioner = &rookCephv1.CephClient{}
-	r.cephClientRBDProvisioner.Name = fmt.Sprintf("%s-%s", "cephclient-rbd-provisioner", r.storageConsumer.Name)
+	cephClientName := &cephClientFormat{StorageConsumerName: r.storageConsumer.Name, ClaimID: "rbd", CephUserType: "provisioner"}
+	r.cephClientRBDProvisioner.Name = generateHashForCephClient(cephClientName)
 	r.cephClientRBDProvisioner.Namespace = r.namespace
+	r.cephClientRBDProvisioner.Annotations["cephUserType"] = fmt.Sprintf("%s-%s-%s", cephClientName.StorageConsumerName, cephClientName.ClaimID, cephClientName.CephUserType)
 
 	r.cephClientRBDNode = &rookCephv1.CephClient{}
-	r.cephClientRBDNode.Name = fmt.Sprintf("%s-%s", "cephclient-rbd-node", r.storageConsumer.Name)
+	cephClientName = &cephClientFormat{StorageConsumerName: r.storageConsumer.Name, ClaimID: "rbd", CephUserType: "node"}
+	r.cephClientRBDNode.Name = generateHashForCephClient(cephClientName)
 	r.cephClientRBDNode.Namespace = r.namespace
+	r.cephClientRBDNode.Annotations["cephUserType"] = fmt.Sprintf("%s-%s-%s", cephClientName.StorageConsumerName, cephClientName.ClaimID, cephClientName.CephUserType)
 
 	r.cephClientCephFSProvisioner = &rookCephv1.CephClient{}
-	r.cephClientCephFSProvisioner.Name = fmt.Sprintf("%s-%s", "cephclient-cephfs-provisioner", r.storageConsumer.Name)
+	cephClientName = &cephClientFormat{StorageConsumerName: r.storageConsumer.Name, ClaimID: "cephfs", CephUserType: "provisioner"}
+	r.cephClientCephFSProvisioner.Name = generateHashForCephClient(cephClientName)
 	r.cephClientCephFSProvisioner.Namespace = r.namespace
+	r.cephClientCephFSProvisioner.Annotations["cephUserType"] = fmt.Sprintf("%s-%s-%s", cephClientName.StorageConsumerName, cephClientName.ClaimID, cephClientName.CephUserType)
 
 	r.cephClientCephFSNode = &rookCephv1.CephClient{}
-	r.cephClientCephFSNode.Name = fmt.Sprintf("%s-%s", "cephclient-cephfs-node", r.storageConsumer.Name)
+	cephClientName = &cephClientFormat{StorageConsumerName: r.storageConsumer.Name, ClaimID: "cephfs", CephUserType: "node"}
+	r.cephClientCephFSNode.Name = generateHashForCephClient(cephClientName)
 	r.cephClientCephFSNode.Namespace = r.namespace
+	r.cephClientCephFSNode.Annotations["cephUserType"] = fmt.Sprintf("%s-%s-%s", cephClientName.StorageConsumerName, cephClientName.ClaimID, cephClientName.CephUserType)
 
 	r.cephClientHealthChecker = &rookCephv1.CephClient{}
 	r.cephClientHealthChecker.Name = fmt.Sprintf("%s-%s", "cephclient-health-checker", r.storageConsumer.Name)
 	r.cephClientHealthChecker.Namespace = r.namespace
+	r.cephClientHealthChecker.Annotations["cephUserType"] = fmt.Sprintf("%s-%s", "cephclient-health-checker", r.storageConsumer.Name)
+
+}
 
+func generateHashForCephClient(cephClientName *cephClientFormat) string {
+	cephClient, err := json.Marshal(cephClientName)
+	if err != nil {
+		panic("failed to marshal")
+	}
+	name := md5.Sum([]byte(cephClient))
+	return hex.EncodeToString(name[:16])
```

Comment:
free-floating util functions (which are not acting on the reconciler) should be positioned at after all reconciler functions